pkg/pillar/pubsub/pubsub.go

```diff
@@ -54,7 +55,7 @@ type SubSyncHandler func(ctx interface{}, synchronized bool)
 // map of agentname, key to get a json string
 // We use StringMap with a RWlock to allow concurrent access.
 type keyMap struct {
-	restartCounter int
+	restartCounter atomic.Int64
```

Comment:
I believe it fixes the test, but I'm curious how it works. As I can see, we have already used a sync mechanism to access this data (the key field is under RW lock). So, how do these two sync mechanisms coexist?

Reply:
I didn't want to introduce another lock for the counter, since I thought it would be an overkill.